### PR TITLE
Add GoalStateV1 support

### PR DESCRIPTION
**Pull request by DavidLiu506**
Add GoadState V1 support to Data Plane Manager, Port Manager and Route Manager and also enable GS version change in configuration file.

### CODE REVIEW

File: services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/ProcessorChainManager.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+@ConditionalOnProperty(prefix = "protobuf.goal-state-message", name = "version", havingValue = "101")
+public class ProcessorChainManager implements IProcessorChainManager {
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorManager.class);
+
```
Comment (cj-chung) [inline services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/ProcessorChainManager.java:32]: @DavidLiu506 Do we need these two functions `ProcessorChainManager` and `ProcessorChainManagerV2`? It seems like both functions has exactly same lines of code.

File: services/route_manager/src/main/java/com/futurewei/alcor/route/controller/NeutronRouterController.java [hunk 2]
```diff
@@ -326,17 +330,24 @@ public RouterInterfaceResponse addInterfaceToNeutronRouter(@PathVariable String
...
-            networkConfiguration.setInternalRouterInfos(internalRouterInfos);
-            List<InternalSubnetEntity> internalSubnetEntities = new ArrayList<>();
-            InternalSubnetEntity internalSubnetEntity = new InternalSubnetEntity(subnetEntity, null);
-            internalSubnetEntities.add(internalSubnetEntity);
-            networkConfiguration.setSubnets(internalSubnetEntities);
-            networkConfiguration.setRsType(Common.ResourceType.NEIGHBOR);
-            this.routerToDPMService.sendInternalRouterInfoToDPM(internalRouterInfo, networkConfiguration);
+            if (version.equals("101")) {
+                InternalRouterInfo internalRouterInfo = this.neutronRouterService.constructInternalRouterInfo(routerid, internalSubnetRoutingTableList);
+                System.out.println("version 101");
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/controller/NeutronRouterController.java:335]: @DavidLiu506  Please remove `System.out.println` here, use `LOG` instead.

File: services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/ProcessorChainManager.java (hunk 1 above)
Comment (DavidLiu506) [reply services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/ProcessorChainManager.java:32]: Yes, AfterProcessor not same, it remove NeighborProcessor. Add RouterProcessor if it move to DPM.

**Comment by cj-chung** [comment]
@DavidLiu506 and @pkommoju can you test this branch in Jenkins?

**Comment by pkommoju** [comment]
> @DavidLiu506 and @pkommoju can you test this branch in Jenkins?

I will start a build shortly.

**Comment by pkommoju** [comment]
> > @DavidLiu506 and @pkommoju can you test this branch in Jenkins?
> 
> I will start a build shortly.

Started.

**Review by cj-chung — APPROVED** [review]
LGTM

**Review by xieus — APPROVED** [review]
This is a really good change.

**Comment by xieus** [comment]
Relink to Issue #720 as it is a pending issue to be resolved later.

**Pull request merged**